File: Chapter05/scala/exercise_twentyeight.scala
val spark = SparkSession
   .builder()
   .appName("exercise_twentysix")
   .getOrCreate()


df.groupby("gender", "country").avg().show(5)

import spark.implicits._
import org.apache.spark.sql.functions.{max, min}
df.groupBy("country", "gender").agg(max(df("salary")), min($"salary"))
 .orderBy($"country".desc)
 .show(7)

File: Chapter05/python/exercise_twentyeight.py
spark = SparkSession \
   .builder \
   .appName("exercise_twentyeight") \
   .getOrCreate()

df.groupby("gender", "country").avg().show(5)

from pyspark.sql import functions as F
df.groupby("title", df["gender"]).agg(F.avg("salary"), F.max("salary")).orderBy(df.title.desc()).show(7)
